app/src/main/kotlin/org/cru/godtools/ui/dashboard/DashboardViewModel.kt

```diff
@@ -66,4 +65,20 @@ class DashboardViewModel @Inject constructor(
         triggerSync()
     }
     // endregion Sync logic
+
+    // region optInNotification logic
+    var permissionStatus: PermissionStatus? = null
+        private set
+
+    fun setPermissionStatus(status: PermissionStatus) {
+        permissionStatus = status
+    }
+
+    private val _showOptInNotification = MutableStateFlow(false)
+    val showOptInNotification: StateFlow<Boolean> = _showOptInNotification
```

Comment:
if you truly want to protect the `StateFlow` from being set elsewhere you can use `_showOptInNotification.asStateFlow()`.

I'm also perfectly fine exposing this property directly as a `MutableStateFlow<Boolean>` and setting it directly instead of going through `setShowOptInNotification()`